src/registry/domain/validators/uploaded-package.js

```diff
@@ -21,9 +19,9 @@ module.exports = function(input){
     return returnError('not_valid');
   }
 
-  const file = input[_.keys(input)[0]];
+  const file = input[0];
 
-  if(file.mimetype !== 'application/octet-stream' || !!file.truncated || file.extension !== 'gz' || file.path.indexOf('.tar.gz') < 0){
```

Comment:
`file.extension` is deprecated and `file.filename` already has the extension check